addons/account/models/account_move.py

```diff
@@ -2671,23 +2671,39 @@ def _get_product_price_and_data(self, product):
                 )
         return product_infos
 
-    def _get_product_catalog_record_lines(self, product_ids, **kwargs):
+    def _get_product_catalog_record_lines(self, product_ids, *, selected_section_id=False, **kwargs):
         grouped_lines = defaultdict(lambda: self.env['account.move.line'])
         for line in self.line_ids:
-            if line.display_type == 'product' and line.product_id.id in product_ids:
+            if (
+                line.section_line_id.id == selected_section_id
+                and line.display_type == 'product'
+                and line.product_id.id in product_ids
+            ):
                 grouped_lines[line.product_id] |= line
         return grouped_lines
 
-    def _update_order_line_info(self, product_id, quantity, **kwargs):
+    def _update_order_line_info(
+            self,
+            product_id,
+            quantity,
+            *,
+            selected_section_id=False,
+            child_field='line_ids',
+            **kwargs,
+        ):
         """ Update account_move_line information for a given product or create a
         new one if none exists yet.
         :param int product_id: The product, as a `product.product` id.
         :param int quantity: The quantity selected in the catalog
+        :param int selected_section_id: The id of section selected in the catalog.
         :return: The unit price of the product, based on the pricelist of the
                  sale order and the quantity selected.
```

Comment:
Why is a mention of "sale order" done here, and why does a function named `_update_order_line_info` return something?